src/timepicker/timepicker.ts

```diff
@@ -158,22 +170,28 @@ export class NgbTimepicker implements ControlValueAccessor {
   }
 
   updateHour(newVal: string | number) {
-    this.model.updateHour(toInteger(newVal) || 0);
-    this.propagateModelChange();
+    if (!this.disabled && !this.readonly) {
+      this.model.updateHour(toInteger(newVal) || 0);
+      this.propagateModelChange();
+    }
   }
 
   updateMinute(newVal: string | number) {
-    this.model.updateMinute(toInteger(newVal) || 0);
-    this.propagateModelChange();
+    if (!this.disabled && !this.readonly) {
+      this.model.updateMinute(toInteger(newVal) || 0);
+      this.propagateModelChange();
+    }
   }
 
   updateSecond(newVal: string | number) {
-    this.model.updateSecond(toInteger(newVal) || 0);
-    this.propagateModelChange();
+    if (!this.disabled && !this.readonly) {
+      this.model.updateSecond(toInteger(newVal) || 0);
+      this.propagateModelChange();
+    }
   }
 
   toggleMeridian() {
-    if (this.meridian) {
+    if (this.meridian && !this.disabled && !this.readonly) {
```

Comment:
I don't think that this check is good. A meridian button should not be active when the whole control is disabled but marking inputs as `readonly` shouldn't prevent people from clicking on a meridian button.

BTW, the same comment applies here as for `update` methods - do we really need any check here? If a button gets disabled users can't click on it anyway...